src/DynamoPublish/ViewModels/PublishViewModel.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Publish.Models;
+using Dynamo.UI.Commands;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Authentication;
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using System.Windows.Input;
+
+namespace Dynamo.Publish.ViewModels
+{
+    public class PublishViewModel : NotificationObject
```

Comment:
No, ViewModel as well as Model can't be internal.
Explanation:
We can bind just public members. So ViewModel should be public.
ViewModel can't be created without Model as parameter in constructor.
Since VM is public it can't take internal members in constructor.
As result Model can't be internal.